### PR TITLE
Add config option to make a domain restriction for google OAuth 

### PR DESCRIPTION
**Pull request by ikewat**
Hi, I currently using it for company level knowledge sharing, and I would like to force users to log in from Google OAuth using our own domain.

So I make a patch on 
```
- passport.authenticate('google', { scope: ['profile'] })(req, res, next)
+ passport.authenticate('google', { scope: ['profile'], hostedDomain: config.google.hostedDomain })(req, res, next)
```
and configuration setups.

I hope that makes sense.

I also update the `yarn.lock` within MR, cause I found there was a build failed when I deploy on Heroku due to `package.json` was updated.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -10637,10 +10637,10 @@ retry-as-promised@^3.2.0:
...
     any-promise "^1.3.0"
 
-reveal.js@~3.7.0:
-  version "3.7.0"
-  resolved "https://registry.yarnpkg.com/reveal.js/-/reveal.js-3.7.0.tgz#7afaf72fd963000381289d58f3aa54c0c46b150c"
-  integrity sha512-HTOTNhF5mQAw6fcsptk4oql/DEEUwTG0YHk/LzTNNx0/3IgvOQZqKzvlK/zNpqqKMLlhn1gH9Nvp+FFoc/e5/w==
+reveal.js@~3.9.2:
+  version "3.9.2"
+  resolved "https://registry.yarnpkg.com/reveal.js/-/reveal.js-3.9.2.tgz#7f63d3dfec338b6c313dcabdf006e8cf80e0b358"
+  integrity sha512-Dvv2oA9FrtOHE2DWj5js8pMRfwq++Wmvsn1EyAdYLC80lBjTphns+tPsB652Bnvep9AVviuVS/b4XoVY9rXHLA==
```
Comment (SuperSandro2000) [inline yarn.lock]: Can you remove this unrelated change?

**Comment by SISheogorath** [comment]
LGTM, but I'll merge it locally to remove the update of the yarn.lock (there will be a broader update happening in master)

**Comment by SISheogorath** [comment]
merged by d5ae45b268452f315926bc4a8ded83e112603d0c
